nomad/state/state_store.go

```diff
@@ -589,7 +576,13 @@ func (s *StateStore) upsertJobImpl(index uint64, job *structs.Job, txn *memdb.Tx
 		job.CreateIndex = existing.(*structs.Job).CreateIndex
 		job.ModifyIndex = index
 		job.JobModifyIndex = index
-		job.Version = existing.(*structs.Job).Version + 1
+
+		// Bump the version unless asked to keep it. This should only be done
+		// when changing an internal field such as Stable. A spec change should
+		// always come with a version bump
+		if !keepVersion {
+			job.Version = existing.(*structs.Job).Version + 1
```

Comment:
I didn't think about this before, but is incrementing here safe? (As if 2 job upserts are based off the same prior version only only of the two will be applied successfully?)